File: SWEA/test_1.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <cmath>
#include <tuple>
using namespace std;

vector <int> DY = { -1,-1,0,0,1,1 };
vector <int> DX = { -1,0,-1,1,0,1 };



pair<int, int> getXY(int n) {
	int diff = 1;
	while (n > 0) {
		int temp = n - diff;
		if (temp > 0) {
			diff++;
			n = temp;
		}
		else {
			int y = diff;
			int x = n;
			return make_pair(y, x);
		}
	}
	// Handle invalid case if needed
	return make_pair(-1, -1);
}

int min_dist(int s, int e) {
	pair<int, int> sxy = getXY(s);
	pair<int, int> exy = getXY(e);
	int ey = 0; int ex = 0; int sy = 0; int sx = 0;
	ey = exy.first; ex = exy.second;
	sy = sxy.first; sx= sxy.second;
	// cout << "End (Y, X) :" << ey << " " << ex << endl;
	// cout << "Start (Y, X) :" << sy << " " << sx << endl;
	int mv_y = ey - sy;
	if (ex == sx) {
		return mv_y;
	}
	else if (ex > sx) {
		int diff_x = ex - sx;
		if (diff_x > mv_y) {
			return diff_x;
		}
		else {
			return mv_y;
		}
	}
	else {
		int diff_x = sx - ex;
		return mv_y + diff_x;
	}

}
int main(int argc, char** argv) {
	int test_case;
	int T;
	int s, e; // 출발, 도착 
	int temp;

	cin >> T;

	for (test_case = 1; test_case <= T; ++test_case) {
		cin >> s >> e;
		if (s > e) { // 무조건 출발 노드가 더 작도록 변경 (굳이 안해도 될것 같긴 한데 그래도)
			temp = s;
			s = e;
			e = temp;
		}
		// int answer = bfs(s, e);
		int answer = min_dist(s, e);
		cout << "#" << test_case << " " << answer << endl;

	}
	return 0;
}

File: SWEA/test_1.py
import sys
sys.stdin = open("input.txt", "r")

import math
import heapq
T = int(input())
DY = [-1,-1,0,0,1,1]
DX = [-1,0,-1,1,0,1]

def get_idx(y, x):
    idx = ((y-1)*y) // 2
    idx += x
    return idx

def get_xy(n):
    diff = 1
    while n:
        temp = n - diff
        if temp > 0:
            diff += 1
            n = temp
        else:
            y = diff
            x = n 
            return y, x

# def bfs(tree, s, e):
def bfs(s, e):
    # ey, ex = tree[e-1]
    # sy, sx = tree[s-1]
    # print(ex, ey, sx, sy)
    ey, ex = get_xy(e)
    sy, sx = get_xy(s)
    print(ex, ey, sx, sy)
    q = [[0, sy, sx]]
    visited = [False for _ in range(e+1)]
    visited[s] = True
    while q:
        cnt, y, x = heapq.heappop(q)
        if y == ey and x == ex:
            return cnt
        for dx, dy in zip(DX, DY):
            nx, ny = x + dx, y + dy
            if nx <= 0 or ny <= 0:
                continue
            if nx > ny:
                continue
            if ny > ey:
                continue
            idx = get_idx(ny, nx)
            if idx > e:
                continue
            
            if visited[idx] == False:
                visited[idx] = True
                heapq.heappush(q, [cnt+1, ny, nx])


    
def get_max_height(n):
    for i in range(n+1):
        num = (i*(i+1)) // 2
        if n <= num:
            return i

def min_dist(s, e):
    ey, ex = get_xy(e)
    sy, sx = get_xy(s)
    print("End : ", ey, ex)
    print("Start : ", sy, sx)
    mv_y = ey - sy
    
    if (ex == sx):
        return mv_y
    
    elif (ex > sx):
        diff_x = ex - sx
        if (diff_x > mv_y):
            return diff_x
        else:
            return mv_y
        
    else:
        diff_x = sx - ex
        return mv_y + diff_x
    
for test_case in range(1, T+1):
    s, e = map(int, input().strip().split(' '))
    if s > e:
        temp = s
        s = e
        e = temp
    # print(f"Start : {s} End : {e}")
    # (1) tree 구조로 블럭 번호에 맞춰서 쌓아주기 #
    tree = []
    y, x = 1, 1
    # max_height = int(math.sqrt(e)) + 1
    # max_height = get_max_height(e)
    # print("max height ", max_height)
    # for height in range(1, max_height+1):
    #     for row in range(1, height+1):
    #         tree.append([height, row])
    # print(tree)
    # (2) bfs를 사용해서 최단 거리 찾아주기 #
    # answer = bfs(tree, s, e)
    # answer = bfs(s, e)
    answer = min_dist(s, e)
    print(f"#{test_case} {answer}")

